pkg/apis/autoscaling/validation/validation.go

```diff
@@ -442,6 +442,14 @@ func validateMetricTarget(mt autoscaling.MetricTarget, fldPath *field.Path) fiel
 		allErrs = append(allErrs, field.Invalid(fldPath.Child("type"), mt.Type, "must be either Utilization, Value, or AverageValue"))
 	}
 
+	if mt.Type == autoscaling.ValueMetricType && mt.Value == nil {
+		allErrs = append(allErrs, field.Invalid(fldPath.Child("value"), mt.Value, "must be set for metric target type Value"))
+	}
+
+	if mt.Type == autoscaling.AverageValueMetricType && mt.AverageValue == nil {
+		allErrs = append(allErrs, field.Invalid(fldPath.Child("averageValue"), mt.AverageValue, "must be set for metric target type AverageValue"))
+	}
+
 	if mt.Value != nil && mt.Value.Sign() != 1 {
```

Comment:
Thanks for the PR, we need to proceed carefully whenever we change validation.

The biggest thing to consider is that we can already have bad data persisted that would fail this validation.

A good first step would be to add nil checks to the controller and return errors instead of panicking. We should backport that change.

We also need to ensure that whatever validation tightening we do doesn't break the ability to make unrelated updates to existing bad objects. Add a unit tests that exercise ValidateHorizontalPodAutoscalerUpdate with every supported source type where the old and new object both have targets with mismatched type/value, and try to update something unrelated (like removing a finalizer or setting an annotation) and ensure the validation passes

Reply:
Thanks @liggitt  following from above, I changed the validation to only check for `ObjectMetricSource` as well as added test cases for re-validating previously bad objects which is overall a smaller change that will only affect the types present in the bug. 
I added nil checks to the controller.